### PR TITLE
show first folder of an account incase of invalid folder id

### PR DESCRIPTION
**Pull request by tahaalibra**
if the folder id is not found then the user will be sent to the first folder of that account

please review @ChristophWurst

### CODE REVIEW

File: js/routecontroller.js [hunk 1]
```diff
@@ -108,9 +108,8 @@ define(function(require) {
 
 			var folder = account.get('folders').get(folderId);
 			if (_.isUndefined(folder)) {
-				// TODO: show first folder of this account
-				_this.default();
-				return;
+				folder = account.get('folders').at(0);
```
Comment (ChristophWurst) [inline js/routecontroller.js]: Note: you could also use [``.first()``](http://backbonejs.org/#Collection-Underscore-Methods), but ``.at(0)`` is also fine as it does the same thing ;-)

**Comment by ChristophWurst** [comment]
Changes look good, I'll test this later! Thanks a lot :-)

**Comment by ChristophWurst** [comment]
@tahaalibra is there any case where this is currently used? (I don't really know how to test it)

**Comment by tahaalibra** [comment]
actually i am not sure if there are in test cases...i tested it with the browser, by simply entering owncloud/apps/mail#accounts/1/folders/(placing some random text)

**Comment by ChristophWurst** [comment]
- [ ] TODO: update changelog

**Comment by tahaalibra** [comment]
i messed up on rebase..i will fix this now

**Comment by ChristophWurst** [comment]
@tahaalibra please rebase on current master, there should only be one commit. Let me know if you need help :)

**Comment by ChristophWurst** [comment]
And please adjust the changelog accordingly, this fix is part of 0.4.5 not 0.4.4 ;-)

**Comment by ChristophWurst** [comment]
:+1: works perfectly! Thanks

**Pull request merged**